traits/trait_types.py

```diff
@@ -1430,6 +1431,21 @@ def validate(self, object, name, value):
 
             Note: The 'fast validator' version performs this check in C.
         """
+        try:
+            # Note: Python 3.5 does not implement __fspath__
```

Comment:
I'd be happy to implement this functionality *just* for Python 3.6 and later, and for Python 3.5 keep the current `File = Str` behaviour. I suspect we have very few users who care about Python 3.5.